decompressed/base/etc/init.d/rootdevice

```diff
@@ -252,56 +229,95 @@ install_gui() {
   echo "  Process done. Your modem is now rooted!" >/dev/console
 }
 
-check_bank() {
-  bank_check=$(uci get -q modgui.var.bank_check)
+is_dual_bank()
+{
+  grep -c "\"bank_2\"" /proc/mtd >/dev/null
+}
+
+check_and_apply_obp() {
+  check_obp=$(uci get -q modgui.var.check_obp)
   bootedbank=$(cat /proc/banktable/booted)
   activebank=$(cat /proc/banktable/active)
-  env_config_bank1="/overlay/bank_1/etc/config/modgui"
-  if [ "$bootedbank" != "$activebank" ]; then
-    echo "  Detected corrupted active bank... Restoring..." >/dev/console
-    restore_bank
-  fi
+  notbooted=$(cat /proc/banktable/notbooted)
+  notbooted_modgui_configfile="/overlay/$notbooted/etc/config/modgui"
 
-  if [ "$bootedbank" == "bank_2" ]; then
-    if [ -f $env_config_bank1 ]; then
-      get_config_bank1=$(< $env_config_bank1 grep bank_check | awk '{print $3}' | sed "s/'//g")
-      if [ -n "$get_config_bank1" ] && [ ! "$(uci -q get modgui.var.bank_check)" ]; then
-        uci set modgui.var.bank_check="$get_config_bank1"
+  if [[ "$bootedbank" != "$activebank" && "$bootedbank" == "bank_2" ]]; then
+    #TODO: add checks to be sure the bank_1 is unbootable (is this really needed?)
+    #TODO: add checks to be sure the bank_1 overlay is SAFE and bootable (empty/prerooted?)
+    echo "  OBP sanity check succeeded continue boot complete..." >/dev/console
+  else
+    #OBP not OK, check if was an user choice otherwise automatically apply (this should be the case called also on first install)
+
+    if [ ! "$(uci -q get modgui.var.check_obp)" ]; then
+      #we must inizialize the check_obp var as is not found
+      if [ -f $notbooted_modgui_configfile ]; then
+        #if we are in a clean bank_2 overlay but in the notbooted bank overlay there was a modgui config file restore var from it
+        get_config_notbooted=$(< $notbooted_modgui_configfile grep check_obp | awk '{print $3}' | sed "s/'//g")
+        if [ -n "$get_config_notbooted" ] && [ ! "$(uci -q get modgui.var.check_obp)" ]; then
+          uci set modgui.var.check_obp="$get_config_notbooted"
+          uci commit modgui
+        fi
```

Comment:
Non ho ben capito come si possa finire in questa condizione ma in generale se trovi roba della gui nel banco non avviato dovresti sempre per sicurezza fare un rm -rf /overlay/$notbooted dopo aver importato quel che cercavi. Altrimenti vorrebbe dire che a seconda di come è valorizzato quel flag, quando vale 0 qui vai ad installare una seconda copia della mod.

Reply:
non so se è giusto mettere un rm dell'overlay notbooted li, in quel caso ci finisci se avevi disattivato l'obp si è rotto il bank che avevi avviato e ti è finito con il failboot all'altro bank, per rispetto del fatto che hai messo la flag di non attivare l'obp non ti tocco l'altro bank e nemmeno l'overlay

Reply:
si ma se c'è la mod installata anche per l'altro banco qui vai a fare una seconda installazione e a riempire l'overlay

più che cancellargli l'overlay forse dovresti abortire e dare errore

Reply:
No, l'installazione non parte: non ci sarebbe la flag force ne il placeholder in root
l'unica cosa che fà è controllarti che ci sia root, la seriale, e il disable del coredump (le funzioni che rimarrebbero anche dopo un factory reset)

Reply:
Se disabilito il check di OBP, metto il firmware nell'altro banco riavvio in quel banco e quel banco **è pulito** come hai scritto nel commento allora l'unico modo che ha questo script di venire eseguito è se viene lanciato manualmente nel tentativo di installare la mod anche su quest'altro banco, e questo va impedito per ovvi motivi. Il caso di cui parli tu in cui invece ci finisci per sbaglio, cioè per failboot, non può capitare perchè per esserci questo script installato ed eseguito automaticamente al boot vorrebbe dire che l'overlay di questo secondo banco **non è pulito** come sarebbe dovuto essere, ma qualcuno ci aveva lasciato dentro rimasugli di mod.

Quindi entri sicuramente nel primo caso quando un utonto tenta la genialata della doppia installazione, mentre non entreresti mai in quel codice nel caso innocuo a cui hai pensato tu, a meno di una situazione di errore dove questo script rootdevice è su entrambi i banchi.

Reply:
ipotizzavo il caso in cui anche il banco notbooted aveva l'overlay con il pacchetto base e che quindi glielo avrebbe solo rootato. Piu che utonto io chi andava a disattivare l'OBP per avere entrambi i banchi runnabili a sto punto lo avevo visto come utente consapevole (un po come chi ora va a disattivare il bank_check per poter fare l'OBP)

per di piu attualmente l'installazione delle GUI su entrambi gli overlay è possibile, i danni si fanno se si iniziano ad installare anche le estensioni (e ovviamente sempre e solo sui device con solo 64mb)